### PR TITLE
T000 - Talktorial template

### PR DESCRIPTION
**Pull request by dominiquesydow**
## Details

- Talktorial ID: T000
- Title: Talktorial template
- Original authors: @dominiquesydow 
- Reviewer(s): @AndreaVolkamer 
- Date of review: 15.12.2020

## TODO

Update template w.r.t. @AndreaVolkamer's suggestions (that she send via PM - copied here):

- [x] (it is self-explanatory, but one could add a note at the beginning that examples are taken from T001)
- [x] Practical: Short summay -> Short summary
- [x] In the path description make it even more explicit, that no relative paths should be given 
- [x] Info for imports (copy info on ordering from issue (?))
- [x] maybe for later: generally we could add some general tips from the PR description in here, I will list a few here
  - [ ] Style guides black, Pep8, …
  - [x] If functions are used, use docstrings, copy one example over? (one more comment on fcts (local/global variables - do not use globals in function) + link to docstring documentation
  - [x] Or tips at the end: 
    - Clear output and rerun, does it finish without errors?
    - Deterministic behavior/output? (help with CI comments)
    - Check runtime?
- [x] In talktorial practical header repeat once that one should make sure the titles are the same as above (as well as the indentation)

### CODE REVIEW

**Comment by dominiquesydow** [comment]
Hi @AndreaVolkamer ,

This PR is ready for your review.
https://github.com/volkamerlab/teachopencadd/blob/ds-t000-revamp/teachopencadd/talktorials/T000_template/talktorial.ipynb

I did not add an general comment on

> Style guides black, Pep8, …

since I think that Jaime or Jeff wanted to set up a document on best practices anyways, which we will be able to link here in the future.

I did referring though to (a) specific PEP8 guidelines e.g. for imports and (b) our template issue.

File: teachopencadd/talktorials/T000_template/talktorial.ipynb [hunk 1]
```diff
@@ -12,13 +34,31 @@
     "- First and last name, year(s) of contribution, lab, institution"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "_The examples used in this talktorial template are taken from __Talktorial T001__._"
```
Comment (AndreaVolkamer) [inline teachopencadd/talktorials/T000_template/talktorial.ipynb:41]: maybe include an actual link to T001?

File: teachopencadd/talktorials/T000_template/talktorial.ipynb [hunk 2]
```diff
@@ -73,24 +135,60 @@
...
     "Figure and description taken from: [<i>Nucleic Acids Res.</i> (2015), <b>43</b>, 612-620](https://academic.oup.com/nar/article/43/W1/W612/2467881)."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-block alert-info\">\n",
+    "    \n",
+    "<b>Figures</b>: Place images in the <code>images/</code> folder and include them using Markdown <code>![Figure title](images/some_figure.jpg)</code>. Add a figure caption in the format provided below.\n",
```
Comment (AndreaVolkamer) [inline teachopencadd/talktorials/T000_template/talktorial.ipynb:187]: .. in the format provided **above** (not below)?

File: teachopencadd/talktorials/T000_template/talktorial.ipynb [hunk 3]
```diff
@@ -159,6 +326,22 @@
...
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-block alert-info\">\n",
+    "\n",
+    "<b>Relative paths</b>: Please do not use relative paths to example input/output data or images.\n",
+    "Instead add this talktorial's path to the global <code>HERE</code>. Define all paths relative to this path.\n",
+    "\n",
+    "<br/>\n",
+    "    \n",
+    "If your talktorial has input/output data, please define the global <code>DATA</code>, which points to this talktorial's data folder (check out the default folder structure of each talktorial).\n",
```
Comment (AndreaVolkamer) [inline teachopencadd/talktorials/T000_template/talktorial.ipynb:340]:
```suggestion
    "<b>Relative paths</b>: Please define all paths relative to this talktorial's path by using the global variable <code>HERE</code> and define all paths relative to this path.\n",
    "\n",
    "<br/>\n",
    "    \n",
    "E.g. if your talktorial has input/output data, please define the global <code>DATA</code>, which points to this talktorial's data folder (check out the default folder structure of each talktorial).\n",
```

Comment (AndreaVolkamer) [reply teachopencadd/talktorials/T000_template/talktorial.ipynb:340]: Just a suggestion, I felt the before sentences were somehow duplicated.

File: teachopencadd/talktorials/T000_template/talktorial.ipynb [hunk 4]
```diff
@@ -179,15 +362,63 @@
...
+    "    logp = Descriptors.MolLogP(molecule)\n",
+    "    # Ro5 conditions fulfilled?\n",
+    "    conditions = [molecular_weight <= 500, n_hba <= 10, n_hbd <= 5, logp <= 5]\n",
+    "    ro5_fulfilled = sum(conditions) >= 3\n",
+    "    # Return True if no more than one out of four conditions is violated\n",
+    "    return pd.Series(\n",
+    "        [molecular_weight, n_hba, n_hbd, logp, ro5_fulfilled],\n",
+    "        index=[\"molecular_weight\", \"n_hba\", \"n_hbd\", \"logp\", \"ro5_fulfilled\"],\n",
+    "    )"
+   ]
```
Comment (AndreaVolkamer) [inline teachopencadd/talktorials/T000_template/talktorial.ipynb:421]: Could we add before/after the example function, to use meaningful function and parameter names?

File: teachopencadd/talktorials/T000_template/talktorial.ipynb [hunk 5]
```diff
@@ -238,6 +471,30 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"alert alert-block alert-info\">\n",
+    "\n",
+    "<b>Useful checks at the end</b>: \n",
+    "    \n",
+    "<ul>\n",
+    "<li>Clear output and rerun your runtime. Does it finish without errors?</li>\n",
```
Comment (AndreaVolkamer) [inline teachopencadd/talktorials/T000_template/talktorial.ipynb:484]:
```suggestion
    "<li>Clear output and rerun your complete notebook. Does it finish without errors?</li>\n",
```

**Review by AndreaVolkamer — COMMENTED** [review]
@dominiquesydow awesome, I really like the added information and the highlighted boxes!
I added only very few minor suggestions, feel free to include them. The notebook can afterwards be merged without further approval from my side.

**Comment by dominiquesydow** [comment]
Great, thanks, @AndreaVolkamer - all your suggestions are integrated. Will merge now.

**Pull request merged**